arch/x86/pagetables.c

```diff
@@ -420,6 +421,18 @@ void *vmap_4k(cr3_t *cr3_ptr, void *va, mfn_t mfn, unsigned long l1_flags,
     return va;
 }
 
+void *vmap_pmm_refill_nolock(void *va, mfn_t mfn, unsigned long l1_flags) {
+    unsigned long _va = _ul(va) & PAGE_ORDER_TO_MASK(PAGE_ORDER_4K);
+
+    dprintk("%s: va: 0x%p mfn: 0x%lx\n", __func__, va, mfn);
+
+    BUG_ON(!vmap_lock);
+    va = _vmap_4k(&cr3, _ptr(_va), mfn, l1_flags, false);
+    BUG_ON(!vmap_lock);
```

Comment:
Maybe these BUG_ONs should be asserts instead? Please also add a comment why they are there.